src/Model/Pane.re

```diff
@@ -0,0 +1,28 @@
+/*
+ * Pane.re
+ *
+ * State tracking the bottom 'UI' pane
+ */
+
+[@deriving show({with_path: false})]
+type paneType =
+  | Search
+  | Diagnostics;
+
+type t = {
+  activePane: paneType,
+  isOpen: bool,
+};
+
+let initial = {activePane: Search, isOpen: false};
+
+let getType = pane => !pane.isOpen ? None : Some(pane.activePane);
+
+let isOpen = pane => pane.isOpen;
+
+let isTypeOpen = (paneType, pane) =>
+  pane.isOpen && pane.activePane == paneType;
+
+let hide = pane => {...pane, isOpen: false};
+
+let show = paneType => {activePane: paneType, isOpen: true};
```

Comment:
This confused me a bit because I expected `show` to return a string as it usually does. It might be an idea to call these something like `setOpen` and `close` until we're weaned off the `show` functions.

Reply:
Oh, good point, that is confusing. I've switched it to `setOpen` and `setClosed` :